lighthouse-core/gather/driver/wait-for-condition.js

```diff
@@ -258,6 +262,69 @@ function waitForCPUIdle(session, waitForCPUQuiet) {
   };
 }
 
+/* c8 ignore start */
+
+/**
+ * This function is executed in the page itself when the document is first loaded.
+ *
+ * Used by _waitForCPUIdle and executed in the context of the page, updates the ____lastLongTask
+ * property on window to the end time of the last long task.
+ */
+function registerPerformanceObserverInPage() {
+  // Do not re-register if we've already run this script.
+  if (typeof window.____lastLongTask !== 'undefined') return;
+
+  window.____lastLongTask = window.__perfNow();
+  const observer = new window.PerformanceObserver(entryList => {
+    const entries = entryList.getEntries();
+    for (const entry of entries) {
+      if (entry.entryType === 'longtask') {
+        const taskEnd = entry.startTime + entry.duration;
+        window.____lastLongTask = Math.max(window.____lastLongTask || 0, taskEnd);
+      }
+    }
+  });
+
+  observer.observe({entryTypes: ['longtask'], buffered: true});
```

Comment:
as is the buffered flag which didn't exist when we originally wrote this when PerfObservers were brand new hotness :)

Reply:
this seems like a big fix?

Reply:
not really that big. we previously installed it before even the document is available and we always assume that a longtask *just* happened so the likelihood of this catching anything new is very small.

now that we're registering slightly later in the page's lifecycle it makes more semantic sense, but again because of our assumption a longtask *just* happened on installation it doesn't make much of a difference in practice